### ISSUE TITLE
Tags not added on build creation via .NET API

### ISSUE THREAD

**Issue by jasonbert** [MicrosoftDocs/vsts-rest-api-specs#83]
I'm using the following code to create a new build:

```
var buildToQueue = new Build
{
	Definition = new DefinitionReference
	{
		Id = pullRequestBuildInfo.BuildDefinitionId
	},
	Project = project,
	SourceBranch = pullRequestBuildInfo.SourceBranch,
	SourceVersion = pullRequestBuildInfo.CommitId,
	Reason = BuildReason.PullRequest,
	Parameters = "{\"PullRequestId\":\"" + pullRequestBuildInfo.Id + "\"}"
};

var buildTags = new List<string>
{
	$"Source - {pullRequestBuildInfo.SourceBranch}",
	$"Target - {pullRequestBuildInfo.TargetBranch}",
	$"Author - {pullRequestBuildInfo.Author}"
};

buildToQueue.Tags.AddRange(buildTags);

var queuedBuild = buildClient.QueueBuildAsync(buildToQueue);
```

My expectation is that the tags would be created and show on the build, however they aren't. Do I have to call the following after queuing the build to then add the tags?

`buildClient.AddBuildTagsAsync(buildTags, pullRequestBuildInfo.BuildProjectId, queuedBuild.Id);`

**Comment by chrisrpatterson** [comment]
Yes I believe you need to add tags once the build exists.  @scottdallamura can you verify?

**Comment by jasonbert** [comment]
Shame, thanks for considering putting something on the backlog though :)

**Comment by SiddhanthUnnithan** [comment]
Hi there,

Adding to this. I spent a couple of hours last Friday trying to add tags to a build object prior to queuing it. It wasn't clear why there was a hard requirement for the build to exist before adding a tag to it. I was able to workaround this problem by waiting for the build to be created and then asynchronously adding a tag:

```C#
// mostly pseudo-code
result = await client.QueueBuildAsync(buildObj);

// add tag
await client.AddBuildTagAsync(project, result.Id, "tag");
```

It would've been great if I could add the tag directly to _buildObj_ prior to making the _queueBuildAsync_ call, similar to what @jasonbert had in his original comment. Has there been any conversation around moving this out of the Backlog?

Best,
Sid
